src/shared/interface/web/react/project/state/project-actions.ts

```diff
@@ -0,0 +1,8 @@
+import { Project } from "@/shared/entities/project";
+
+export type Action =
+  | { type: 'FETCH_START' }
+  | { type: 'FETCH_SUCCESS'; payload: Project[] }
```

Comment:
this feels weird to me, project actions immediately strike as something performed on the project, whereas fetch success / error are an action consequence. is there a deeper reason behind why we do it this way? is there an easy way to partition this / should we?

Reply:
okay that makes sense. did you consider naming the actions more transition-like? maybe PROJECT_SELECTED and PROJECT_CREATED would make more sense in this context?